pkg/handler/processor/guesser/format_guesser.go

```diff
@@ -35,9 +36,11 @@ var (
 	documentFormatGuessers = map[string]DocumentFormatGuesser{}
 )
 
-func RegisterDocumentFormatGuesser(g DocumentFormatGuesser, name string) {
+func RegisterDocumentFormatGuesser(g DocumentFormatGuesser, name string) error {
 	if _, ok := documentFormatGuessers[name]; ok {
-		logrus.Warnf("the document type guesser is being overwritten: %s", name)
+		fmt.Printf("the document type guesser is being overwritten: %s", name)
```

Comment:
same with this one. Just return an error and upstream we can check if it is getting "overwritten" and log there.